version: '3'

services:

    redis:
        image: redis
        ports:
            - '6379:6379'
        volumes:
            - redis_data:/data
        env_file:
            - .env

    db:
        image: postgres
        env_file:
            - .env
        volumes:
            - pg_data:/var/lib/postgresql/data
        expose:
            - '5432'
        healthcheck:
            test: [ "CMD-SHELL", "pg_isready -U postgres" ]
            interval: 10s
            timeout: 5s
            retries: 5

    backend:
        build:
            context: .
            dockerfile: Dockerfile
        container_name: backend
        command: >
            sh -c "python manage.py makemigrations && python manage.py migrate && gunicorn django_avito.wsgi:application --bind 0.0.0.0:8000"
        ports:
            - "8000:8000"
        depends_on:
            - db
            - redis
        restart: always

    celery:
        build: .
        tty: true
        command: celery -A django_avito worker -l INFO
        depends_on:
            - redis
            - backend

    celery_beat:
        build: .
        tty: true
        restart: always
        command: celery -A django_avito beat -l info -S django
        depends_on:
            - redis
            - backend

    nginx:
        image: nginx:latest
        container_name: nginx
        volumes:
            - ./nginx.conf:/etc/nginx/nginx.conf:ro
            - ./certs:/etc/nginx/certs:ro
        ports:
            - "443:443"
        depends_on:
            - backend
        restart: always

volumes:
    pg_data:
